contracts/SwapAndCall.sol

```diff
@@ -0,0 +1,80 @@
+pragma solidity ^0.5.0;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import 'hardlydifficult-ethereum-contracts/contracts/proxies/CallContract.sol';
+
+
+/**
+ * @notice call any contract(s) and then refund the remaining balance (if any)
+ * Idea from https://etherscan.io/address/0x11111254369792b2Ca5d084aB5eEA397cA8fa48B#code
+ */
+contract SwapAndCall
+{
+  using CallContract for address;
+  using SafeERC20 for IERC20;
+
+  // Accept ETH from other contracts
+  function() external payable {}
+
+  /**
+   * @notice Make a series of calls and then refunds any funds remaining.
+   * @param _contractsAndTokensToRefund The contract address for each call to make.
+   * Followed by the address of any ERC-20 tokens to refund at the end.
+   * ETH is always refunded.  If no additional token is required this may be empty.
+   * @param _callDataConcat Call data for each call, appended into a single bytes field.
+   * @param _startPositionsAndValues The start position of callData for each call after
+   * the first (assumed to be 0).
+   * Followed by the call value to include with each call.
+   * @dev Variables are reused to avoid a `stack too deep` error.
+   */
+  function swapAndCall(
+    address[] calldata _contractsAndTokensToRefund,
+    bytes calldata _callDataConcat,
+    uint[] calldata _startPositionsAndValues
```

Comment:
It's not clear to me from the param comment how these values are ordered. At first I thought they were alternating like so:  [startPosition1, value1, startPosition2, value2, etc...], but now I'm thinking maybe it's actually scenario 2:[startPosition1, startPosition2, etc..., value1,, value2, etc...] ?

Reply:
it's pos1, pos2, value0, value1, value2

Reply:
Ahh. ok, makes sense.